### PR TITLE
ddl: introduce DDLReorgMeta.ReorgType to persist backfill way

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/index.go [hunk 1]
```diff
@@ -680,29 +670,34 @@ func (w *worker) onCreateIndex(d *ddlCtx, t *meta.Meta, job *model.Job, isPK boo
...
 	if IsEnableFastReorg() {
-		if lightning.GlobalEnv.IsInited {
-			err := lightning.BackCtxMgr.Register(ctx, indexInfo.Unique, job.ID, job.ReorgMeta.SQLMode)
+		piTREnabled := isPiTREnable(w)
+		if lightning.GlobalEnv.IsInited && !piTREnabled {
+			err := lightning.BackCtxMgr.Register(w.ctx, indexInfo.Unique, job.ID, job.ReorgMeta.SQLMode)
 			if err != nil {
-				return // Txn backfill
+				logutil.BgLogger().Info("fallback to default backfill process", zap.Error(err))
+				job.ReorgMeta.ReorgTp = model.ReorgTypeTxn
```
Comment (Benjamin2037) [inline ddl/index.go:686]: Once IsEnableFastReorg， even backend is not create successful, it could goto ReorgTypeTxnMerge

Comment (Benjamin2037) [inline ddl/index.go:686]:
```suggestion
				job.ReorgMeta.ReorgTp = model.ReorgTypeTxnMerge
```

Comment (tangenta) [reply ddl/index.go]: Done

File: ddl/lightning/backend.go [hunk 2]
```diff
@@ -106,8 +106,3 @@ func (bc *BackendContext) Flush(engineKey string) error {
 func (bc *BackendContext) NeedRestore() bool {
```
Comment (Benjamin2037) [inline ddl/lightning/backend.go:106]: Seems this function is usless.

Comment (tangenta) [reply ddl/lightning/backend.go]: Removed.

File: ddl/index.go [hunk 3]
```diff
@@ -680,29 +670,32 @@ func (w *worker) onCreateIndex(d *ddlCtx, t *meta.Meta, job *model.Job, isPK boo
...
-//  1. Txn backfill(original): all the index KVs are written through the transaction interface.
-//  2. Lightning backfill: the index KVs are encoded to SST files and imported to the storage directly.
-//     The incremental index KVs written by DML are redirected to a temporary index.
-//     After the backfill is finished, the temporary index records are merged back to the original index.
-//  3. Txn-merge backfill: the backfill index KVs are written through the transaction interface.
-//     The incremental index KVs written by DML are redirected to a temporary index.
-//     After the backfill is finished, the temporary index records are merged back to the original index.
-func checkAndInitLightningBackfillCtx(ctx context.Context, job *model.Job, indexInfo *model.IndexInfo) {
+// pickBackfillProcess determines which backfill process will be used.
+func pickBackfillProcess(w *worker, job *model.Job, indexInfo *model.IndexInfo) model.ReorgType {
```
Comment (Benjamin2037) [inline ddl/index.go:674]: When crash and restart job,  seems there is no this part logic in this pr. you will do it later?

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Pull request merged**